### PR TITLE
[Debugger] Add debug configurations guide

### PR DESCRIPTION
**Pull request by praveennadarajah**
## Purpose
This PR will add debug configurations guide to the existing `Run and Debug` guide.

### CODE REVIEW

File: learn/tooling-guide/visual-studio-code-extension/run-and-debug/run-and-debug.md [hunk 1]
```diff
@@ -177,6 +177,56 @@ The Ballerina VSCode debugger lets you evaluate expressions in the ways below.
...
+
+Here are the default configurations generated for the ballerina debugging:
+
+![Debug Configurations](/learn/images/debug-configurations.png)
+
+<br/>
+
+>**Info:** You can debug a simple ballerina program even without the `launch.json` configurations file, but it is not possible to manage launch configurations and set up advanced debugging.
+
+The auto-generated `launch.json` file consists of three main configurations, namely, `Ballerina Debug`, `Ballerina Test` and `Ballerina Remote`. Each configuration supports different attributes, and those attributes can be identified with the help of IntelliSense suggestions (`⌃Space`).
```
Comment (NipunaRanasinghe) [inline learn/tooling-guide/visual-studio-code-extension/run-and-debug/run-and-debug.md:204]:
```suggestion
```

File: learn/tooling-guide/visual-studio-code-extension/run-and-debug/run-and-debug.md [hunk 2]
```diff
@@ -177,6 +177,56 @@ The Ballerina VSCode debugger lets you evaluate expressions in the ways below.
...
+
+### Ballerina launch.json attributes
+
+There are many `launch.json` attributes to help and support different types of ballerina debugging scenarios. All the available attributes can be identified with the help of IntelliSense suggestions (`⌃Space`).
```
Comment (NipunaRanasinghe) [inline learn/tooling-guide/visual-studio-code-extension/run-and-debug/run-and-debug.md:208]:
```suggestion
The auto-generated `launch.json` file consists of three main configurations, namely, `Ballerina Debug`, `Ballerina Test` and `Ballerina Remote`. Each configuration supports different attributes, and those attributes can be identified with the help of IntelliSense suggestions (`⌃Space`).
```

File: learn/tooling-guide/visual-studio-code-extension/run-and-debug/run-and-debug.md [hunk 3]
```diff
@@ -177,6 +177,56 @@ The Ballerina VSCode debugger lets you evaluate expressions in the ways below.
...
+- `programArgs` - Any program arguments that are required to be passed into the `main` function of the Ballerina program to be launched, can be passed as a list of strings.
+- `commandOptions` - If required, you can configure command options for the Ballerina program to be launched, as a list of strings. You can see the list of all the available command options by executing the following CLI commands in your terminal.
+    - For `Ballerina Debug` configuration - `ballerina run --help`.
+    - For `Ballerina Test` configuration - `ballerina test --help`.
+- `env` - Any environment variables you need to configure for the launching Ballerina program can be passed as a map of strings (name and value).
+- `debugTests` - Indicates whether to debug the tests for the given script.
+
+The following attributes are supported for all Ballerina `attach` configurations.
+
+- `debuggeeHost` - remote debug server host.
```
Comment (NipunaRanasinghe) [inline learn/tooling-guide/visual-studio-code-extension/run-and-debug/run-and-debug.md:227]:
```suggestion
- `debuggeeHost` - Host address of the remote process to be attached (If not specified, default value be the localhost(`127.0.0.1`)).
```

File: learn/tooling-guide/visual-studio-code-extension/run-and-debug/run-and-debug.md [hunk 4]
```diff
@@ -177,6 +177,56 @@ The Ballerina VSCode debugger lets you evaluate expressions in the ways below.
...
+- `debuggeePort` - remote debug server port.
```
Comment (NipunaRanasinghe) [inline learn/tooling-guide/visual-studio-code-extension/run-and-debug/run-and-debug.md:228]:
```suggestion
- `debuggeePort` - Port number of the remote process to be attached.
```

File: learn/tooling-guide/visual-studio-code-extension/run-and-debug/run-and-debug.md [hunk 5]
```diff
@@ -177,6 +177,56 @@ The Ballerina VSCode debugger lets you evaluate expressions in the ways below.
...
+
+<br/>
+
+Here are the default configurations generated for the ballerina debugging:
+
+![Debug Configurations](/learn/images/debug-configurations.png)
+
+<br/>
+
+>**Info:** You can debug a simple ballerina program even without the `launch.json` configurations file, but it is not possible to manage launch configurations and set up advanced debugging.
```
[the remark this answers is not in the document]
Comment (praveennadarajah) [reply learn/tooling-guide/visual-studio-code-extension/run-and-debug/run-and-debug.md]: Fixed with https://github.com/ballerina-platform/ballerina-dev-website/pull/2300/commits/61e0fb41df98efaac66ec68f785b2f6abb798190

File: learn/tooling-guide/visual-studio-code-extension/run-and-debug/run-and-debug.md (hunk 1 above)
Comment (praveennadarajah) [reply learn/tooling-guide/visual-studio-code-extension/run-and-debug/run-and-debug.md]: Fixed with https://github.com/ballerina-platform/ballerina-dev-website/pull/2300/commits/61e0fb41df98efaac66ec68f785b2f6abb798190

File: learn/tooling-guide/visual-studio-code-extension/run-and-debug/run-and-debug.md (hunk 2 above)
Comment (praveennadarajah) [reply learn/tooling-guide/visual-studio-code-extension/run-and-debug/run-and-debug.md]: Fixed with https://github.com/ballerina-platform/ballerina-dev-website/pull/2300/commits/61e0fb41df98efaac66ec68f785b2f6abb798190

File: learn/tooling-guide/visual-studio-code-extension/run-and-debug/run-and-debug.md (hunk 3 above)
Comment (praveennadarajah) [reply learn/tooling-guide/visual-studio-code-extension/run-and-debug/run-and-debug.md]: Fixed with https://github.com/ballerina-platform/ballerina-dev-website/pull/2300/commits/61e0fb41df98efaac66ec68f785b2f6abb798190

File: learn/tooling-guide/visual-studio-code-extension/run-and-debug/run-and-debug.md (hunk 4 above)
Comment (praveennadarajah) [reply learn/tooling-guide/visual-studio-code-extension/run-and-debug/run-and-debug.md]: Fixed with https://github.com/ballerina-platform/ballerina-dev-website/pull/2300/commits/61e0fb41df98efaac66ec68f785b2f6abb798190

**Review by praneesha — APPROVED** [review]

**Pull request merged**